src/codemagic/apple/app_store_connect/provisioning/profiles.py

```diff
@@ -120,49 +121,42 @@ def get_bundle_id_resource_id(self, profile: Union[Profile, ResourceId]) -> Link
         """
         https://developer.apple.com/documentation/appstoreconnectapi/get_the_bundle_resource_id_in_a_profile
         """
-        if isinstance(profile, Profile):
+        url = None
+        if isinstance(profile, Profile) and profile.relationships is not None:
             url = profile.relationships.bundleId.links.self
-        else:
+        if url is None:
             url = f'{self.client.API_URL}/profiles/{profile}/relationships/bundleId'
         response = self.client.session.get(url).json()
         return LinkedResourceData(response['data'])
 
-    def list_certificates(self, profile: Union[Profile, ResourceId]) -> List[SigningCertificate]:
+    def list_certificates(self, profile: ResourceId) -> List[SigningCertificate]:
```

Comment:
Signature changes such as this are not backwards compatible and hence breaking. There is no need to introduce them without a good reason as it might have cause errors on consumer side.

This applies to here and elsewhere where similar changes are applied.

Reply:
hmm, no idea why i changed that